actions/marketplace.ts

```diff
@@ -168,6 +170,16 @@ export function installApp(id: string) {
             type: ActionTypes.INSTALLING_MARKETPLACE_ITEM_SUCCEEDED,
             id,
         });
+
+        const callResp = res.data!;
+        if (callResp.type === AppCallResponseTypes.FORM && callResp.form) {
+            dispatch(openAppsModal(callResp.form, creq.context));
+        }
+
+        if (callResp.text) {
+            dispatch(postEphemeralCallResponseForContext(callResp, callResp.text, context));
+        }
+
         return true;
```

Comment:
Can we de-duplicate the code used here? Is there a way to throw the call response in the normal event pipeline?

Reply:
@hanzei We purposely have the logic different in each binding location case, so special cases can easily be handled. We could have a "default" handler to be used that does these sort of actions shown above. A "nothing special" call response handler function

Reply:
I think this is out of scope of this PR though. 

There is an improvement I made in the mobile app for this https://github.com/mattermost/mattermost-mobile/blob/aa6c1ff0583751085967c9fdffd67789f7b24421/app/hooks/apps.ts#L21-L26

```ts
export type UseAppBindingConfig = {
    onSuccess: (callResponse: AppCallResponse, message: string) => void;
    onError: (callResponse: AppCallResponse, message: string) => void;
    onForm?: (form: AppForm) => void;
    onNavigate?: (callResp: AppCallResponse) => void;
}
```

This allows the caller to supply at least `onSuccess` and `onError`, and things like opening a form is done automatically. I made a ticket to add this to the webapp https://mattermost.atlassian.net/browse/MM-50197